File: MediCareWebSpa/src/app/DTOs/request/user-request.dto.ts
export class UserRequestDTO {
  id: number;
  name: string;
  surname: string;
  email: string;
  password: string;
  pesel: string;
  phoneNumber: string;
  specialityId: number | null = null;
  birthDate: Date | null = null;

  constructor(
    id: number,
    name: string,
    surname: string,
    email: string,
    password: string,
    pesel: string,
    phoneNumber: string
  ) {
    this.id = id;
    this.name = name;
    this.surname = surname;
    this.email = email;
    this.password = password;
    this.pesel = pesel;
    this.phoneNumber = phoneNumber;
  }
}

File: MediCareWebApi/DTOs/Request/UserRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace MediCare.DTOs.Request
{
	public class UserRequestDTO : UserBasicDataDTO
	{
		public int Id { get; set; }

		[MaxLength(256, ErrorMessage = "Hasło nie może przekroczyć 256 znaków.")]
		public string? Password { get; set; }

		public int? SpecialityId { get; set; }

		[DataType(DataType.Date, ErrorMessage = "Nieprawidłowy format daty.")]
		public DateTime? BirthDate { get; set; }
	}
}
